neuralmonkey/vocabulary.py

```diff
@@ -536,15 +537,22 @@ def vectors_to_sentences(self,
         Returns:
             List of lists of words.
         """
-        sentences = [[] for _ in
-                     range(vectors[0].shape[0])]  # type: List[List[str]]
+        if isinstance(vectors, list):
+            batch_size = vectors[0].shape[0]
+        elif isinstance(vectors, np.ndarray):
+            batch_size = vectors.shape[1]
+        else:
+            raise TypeError(
+                "Unexpected type of decoder output: {}".format(type(vectors)))
```

Comment:
Podle mě `vectors[0].shape[0]` funguje ve všech případech a nemusí se to tu větvit.

Reply:
Nefunguje v případě, kdy je ten matice má shape (0 × _n_), tak `vectors[0]` neexistuje, ale přesto se z toho dá zjistit batch size.